daemon/init.go

```diff
@@ -0,0 +1,18 @@
+// TODO(ilgooz): remove this file after daemon package made Newable.
+package daemon
+
+import (
+	"log"
+
+	"github.com/mesg-foundation/core/container"
+)
+
+var defaultContainer *container.Container
```

Comment:
Isn't it better to create an instance every time we need it instead of creating it an `init` function ?
It would be good to avoid `init` functions that save a state.
This container is totally stateless so we can re-create it all the time and it will not change anything except some memory/cpu optimization but that are negligible for these features

Reply:
After daemon refactored, `*Container` will be initialized only once and stored in the `*Daemon` struct. Until the refactor we have to initialize a Container type and store it in the package level to be able to use it. I won't recommend _-also don't like-_ initializing Container every time it's needed because it makes additional Docker API calls in the initialization time. Please see [here](https://github.com/mesg-foundation/core/blob/feature/container/container/container.go#L40-L44).

Reply:
ok let's keep it like that for now, and for the initialization of the container every time, you're right with the docker api calls even if we could add a singleton on that or some stuff like that but we will think about this kind of optimization later
So this is ok now :)

Reply:
I directly changed the corresponding commit from the history as you liked. This way we'll not see an irrelevant code in the init.go file.